connectors/sources/sharepoint_online.py

```diff
@@ -354,14 +366,14 @@ async def scroll_delta_url(self, url):
             else:
                 break
 
-    async def _get_json(self, absolute_url):
+    async def _get_json(self, absolute_url, method=HTTPMethod.GET, payload=None):
```

Comment:
It's not `_get_json` any more :D

Maybe worth to split into different methods or just do same as I did in `pipe` (do specific logic in the method calling `self._call_api`)?